main_test.go

```diff
@@ -37,3 +38,78 @@ func TestBuildExporter(t *testing.T) {
 
 	buildExporter(opts)
 }
+
+func TestBuildURI(t *testing.T) {
+
+	tests := []struct {
+		situation   string
+		origin      string
+		newUser     string
+		newPassword string
+		expect      string
+	}{
+		{
+			situation:   "uri with prefix and auth, and auth supplied in opt.User/Password",
+			origin:      "mongodb://usr:pwd@127.0.0.1",
+			newUser:     "xxx",
+			newPassword: "yyy",
+			expect:      "mongodb://usr:pwd@127.0.0.1",
+		},
+		{
+			situation:   "uri with prefix and auth, no auth supplied in opt.User/Password",
+			origin:      "mongodb://usr:pwd@127.0.0.1",
+			newUser:     "",
+			newPassword: "",
+			expect:      "mongodb://usr:pwd@127.0.0.1",
+		},
+		{
+			situation:   "uri with no prefix and auth, and auth supplied in opt.User/Password",
+			origin:      "usr:pwd@127.0.0.1",
+			newUser:     "xxx",
+			newPassword: "yyy",
+			expect:      "usr:pwd@127.0.0.1",
+		},
+		{
+			situation:   "uri with no prefix and auth, no auth supplied in opt.User/Password",
+			origin:      "usr:pwd@127.0.0.1",
+			newUser:     "",
+			newPassword: "",
+			expect:      "usr:pwd@127.0.0.1",
+		},
+		{
+			situation:   "uri with prefix and no auth, and auth supplied in opt.User/Password",
+			origin:      "mongodb://127.0.0.1",
+			newUser:     "xxx",
+			newPassword: "yyy",
+			expect:      "mongodb://xxx:yyy@127.0.0.1",
+		},
+		{
+			situation:   "uri with prefix and no auth, no auth supplied in opt.User/Password",
+			origin:      "mongodb://127.0.0.1",
+			newUser:     "",
+			newPassword: "",
+			expect:      "mongodb://127.0.0.1",
+		},
+		{
+			situation:   "uri with no prefix and no auth, and auth supplied in opt.User/Password",
+			origin:      "127.0.0.1",
+			newUser:     "xxx",
+			newPassword: "yyy",
+			expect:      "mongodb://xxx:yyy@127.0.0.1",
+		},
+		{
+			situation:   "uri with no prefix and no auth, no auth supplied in opt.User/Password",
+			origin:      "127.0.0.1",
+			newUser:     "",
+			newPassword: "",
+			expect:      "127.0.0.1",
```

Comment:
the expectation here is `mongodb://127.0.0.1`, that's how it used to work
```suggestion
			expect:      "mongodb://127.0.0.1",
```

Reply:
my mistake, have correct it both in `main.go` and `main_test.go`

Reply:
it does need a `mongod://` prefix in all situation, according to `https://www.mongodb.com/docs/manual/reference/connection-string/`